### PR TITLE
Update mean example test (#1730)

### PR DESCRIPTION
**Pull request by elgorwi**
- Cleanup and remove python 2 statements
- Change to  'async def' syntax (#1730 )
- Add 'mean_range_test'

### CODE REVIEW

File: examples/mean/tests/test_mean.py [hunk 1]
```diff
@@ -21,72 +17,80 @@ class StreamBusMonitor(BusMonitor):
     _signals = ["valid", "data"]
 
     @cocotb.coroutine
```
Comment (ktbarrett) [inline examples/mean/tests/test_mean.py:19]: These decorators can be removed.

File: examples/mean/tests/test_mean.py [hunk 2]
```diff
@@ -21,72 +17,80 @@ class StreamBusMonitor(BusMonitor):
...
-            yield ReadOnly()
+            await RisingEdge(self.clock)
+            await ReadOnly()
             if self.bus.valid.value:
                 self._recv(int(self.bus.data.value))
 
 
 @cocotb.coroutine
-def clock_gen(signal, period=10, units='ns'):
+async def clock_gen(signal, period=10, units='ns'):
```
Comment (ktbarrett) [inline examples/mean/tests/test_mean.py:31]: This entire thing should be replaced with `Clock`.

File: examples/mean/tests/test_mean.py [hunk 3]
```diff
@@ -21,72 +17,80 @@ class StreamBusMonitor(BusMonitor):
...
     while True:
         signal <= 0
-        yield Timer(period/2, units=units)
+        await Timer(period/2, units=units)
         signal <= 1
-        yield Timer(period/2, units=units)
+        await Timer(period/2, units=units)
 
 
 @cocotb.coroutine
```
Comment (ktbarrett) [inline examples/mean/tests/test_mean.py:39]: And remove this here too.

File: examples/mean/tests/test_mean.py [hunk 4]
```diff
@@ -99,30 +103,33 @@ def mean_randomised_test(dut):
...
 
-    data_width = int(dut.DATA_WIDTH.value)
-    bus_width = int(dut.BUS_WIDTH.value)
+    DATA_WIDTH = int(dut.DATA_WIDTH.value)
+    BUS_WIDTH = int(dut.BUS_WIDTH.value)
     dut._log.info('Detected DATA_WIDTH = %d, BUS_WIDTH = %d' %
-                  (data_width, bus_width))
+                  (DATA_WIDTH, BUS_WIDTH))
 
     cocotb.fork(clock_gen(dut.clk, period=CLK_PERIOD_NS, units='ns'))
```
Comment (ktbarrett) [inline examples/mean/tests/test_mean.py:111]: This should be replaced with `Clock` as well.

File: examples/mean/tests/test_mean.py (hunk 1 above)
Comment (elgorwi) [reply examples/mean/tests/test_mean.py:20]: Done

File: examples/mean/tests/test_mean.py (hunk 2 above)
Comment (elgorwi) [reply examples/mean/tests/test_mean.py]: Done

File: examples/mean/tests/test_mean.py (hunk 4 above)
Comment (elgorwi) [reply examples/mean/tests/test_mean.py]: Done

File: examples/mean/tests/test_mean.py (hunk 1 above)
Comment (ktbarrett) [reply examples/mean/tests/test_mean.py:20]: Was there any issue with changing this?

Comment (elgorwi) [reply examples/mean/tests/test_mean.py]: This is now fixed

**Pull request merged**